src/PowerShellEditorServices/Analysis/AnalysisService.cs

```diff
@@ -515,22 +527,25 @@ private void LogAvailablePssaFeatures()
             return diagnosticRecords;
         }
 
-        private PowerShellResult InvokePowerShell(string command, IDictionary<string, object> paramArgMap)
+        private PowerShellResult InvokePowerShell(string command, IDictionary<string, object> paramArgMap = null)
```

Comment:
the default param could be an empty dictionary and then you don't need he null check on line 536

Reply:
Yes I made this change because one of the calls to this method had to create a new empty dictionary just to call the method. A null check is much cheaper than creating a new dictionary.

I think the question boils down to "do we make the caller or the callee handle the complexity". Since a caller could always pass in `null` for the dictionary, I think defaulting to `null` and making the callee handle it is probably more efficient, safer and easier to use.